overlord/state/state_test.go

```diff
@@ -546,6 +547,17 @@ func (ss *stateSuite) TestEmptyStateDataAndCheckpointReadAndSet(c *C) {
 
 	// no crash
 	st2.Set("a", 1)
+
+	// ensure all maps of state are correctly initialized by ReadState
+	val := reflect.ValueOf(st2)
+	typ := reflect.TypeOf(val.Elem())
+	for i := 0; i < typ.NumField(); i++ {
+		f := typ.Field(i)
+		if f.Type.Kind() == reflect.Map {
+			fv := val.Elem().Field(i)
```

Comment:
maybe to catch the issue we should change the code to do this:

```
	typ := val.Elem().Type()
	var maps []string
	for i := 0; i < typ.NumField(); i++ {
		f := typ.Field(i)
		if f.Type.Kind() == reflect.Map {
			maps = append(maps, f.Name)
			fv := val.Elem().Field(i)
			c.Check(fv.IsNil(), Equals, false, Commentf("Map field %s of state was not initialized by ReadState", f.Name))
		}
	}
	c.Check(maps, DeepEquals, []string{"data"...})
```

Reply:
added this, it's a really nice suggestion!